paasta_tools/cli/cmds/rollback.py

```diff
@@ -193,36 +252,48 @@ def paasta_rollback(args):
         )
         return 1
 
-    git_shas = get_git_shas_for_service(service, deploy_groups, soa_dir)
+    versions = get_versions_for_service(service, deploy_groups, soa_dir)
     commit = args.commit
+    image_version = args.image_version
+    new_version = DeploymentVersion(sha=commit, image_version=image_version)
     if not commit:
         print("Please specify a commit to mark for rollback (-k, --commit).")
-        list_previous_commits(
-            service, deploy_groups, bool(given_deploy_groups), git_shas
+        list_previous_versions(
+            service, deploy_groups, bool(given_deploy_groups), versions
         )
         return 1
-    elif commit not in git_shas and not args.force:
-        print(PaastaColors.red("This Git SHA has never been deployed before."))
+    elif new_version not in versions and not args.force:
+        print(
+            PaastaColors.red(
+                f"This version {new_version} has never been deployed before."
+            )
+        )
         print("Please double check it or use --force to skip this verification.\n")
-        list_previous_commits(
-            service, deploy_groups, bool(given_deploy_groups), git_shas
+        list_previous_versions(
+            service, deploy_groups, bool(given_deploy_groups), versions
         )
         return 1
 
     returncode = 0
 
     for deploy_group in deploy_groups:
-        rolled_back_from = get_currently_deployed_sha(service, deploy_group)
+        rolled_back_from = get_currently_deployed_version(service, deploy_group)
         returncode |= mark_for_deployment(
-            git_url=git_url, service=service, deploy_group=deploy_group, commit=commit
+            git_url=git_url,
+            service=service,
+            deploy_group=deploy_group,
+            commit=commit,
+            image_version=image_version,
         )
 
         # we could also gate this by the return code from m-f-d, but we probably care more about someone wanting to
         # rollback than we care about if the underlying machinery was successfully able to complete the request
-        if rolled_back_from != commit:
+
+        # TODO: Eventually we should include full version in rollback event, but currently leave things as status-quo
```

Comment:
are we not including the full version in the event? I see you changing both `rolled_back_from` and `rolled_back_to` to the new style (`version` rather than `sha`)

Reply:
Oh, good call --  I think in a previous iteration I conditionally set `rolled_back_to/from` to the commit or the version before I decided to just override DeployVersion's `__repr__`. I will remove this misleading comment.